packages/xstate-react/src/useMachine.ts

```diff
@@ -194,7 +203,11 @@ export function useMachine<
           currentState.changed === undefined &&
           Object.keys(currentState.children).length;
 
-        if (currentState.changed || initialStateChanged) {
+        if (
+          (currentState.changed || initialStateChanged) &&
+          // Only change the state if it is not filtered out
+          (!filter || filter?.(currentState, currentState.history!))
```

Comment:
Is TS really requiring `?.` after the `!filter` check?

Reply:
It can be set but might be with diff signature, like `undefined` or something.